elastic/kibana
# [Mappings Editor] Add support for synthetic _source

Closes https://github.com/elastic/kibana/issues/198621

## Summary

This PR adds support for the synthetic _source field in the mappings Advanced options.

Stored option:
<img width="1184" alt="Screenshot 2024-11-14 at 19 19 19" src="https://github.com/user-attachments/assets/086f7a3e-9ca1-42de-9f9c-d3599d839ccf">

Synthetic option selected:
<img width="1184" alt="Screenshot 2024-11-14 at 19 19 27" src="https://github.com/user-attachments/assets/3700bced-212a-4378-b51a-ab7a0f4f7b99">

Disabled option selected:
<img width="1184" alt="Screenshot 2024-11-14 at 19 19 36" src="https://github.com/user-attachments/assets/c7ddcbae-7c78-4477-824e-99b144a1f750">



https://github.com/user-attachments/assets/399d0f95-a5dd-4874-bb8c-e95d6ed38465




How to test:
1. Start Es with `yarn es snapshot --license` (we need Enterprise license to see the Synthetic source option) and Kibana with `yarn start`
2. Go to Index templates/Component templates and start creating a template
3. At the Mappings step, go to Advanced options.
4. Verify that selecting a _source field option translates to the correct Es request.
5. In Index templates form, verify that the default _source option depends on the index mode selected in the Logistics step. For LogsDB and Time series index mode, the default should be synthetic mode; otherwise, the stored option.
6. Verify that in Basic license, the synthetic option is not displayed.



### Checklist
